paddle/majel/allocation.cc

```diff
@@ -0,0 +1,97 @@
+#include <boost/variant.hpp>
+
+#include "allocation.h"
+#include "hl_cuda.h"
```

Comment:
I think we should create malloc.{h,cc}, which calls C runtime `malloc` and `cudaMalloc`, but doesn't depend on hl_cuda.  In that way, we don't make Majel depend on Paddle.